test/uts/controller_test/controller_test.go

```diff
@@ -0,0 +1,379 @@
+	newPodName                    string
+	isErrorExpectedAtFakeFileRead bool
+	expectedCpus                  []string
+}{
+	{"pod_exc", "pod_shared", false, []string{"9-12,15,17"}},
+	{"pod_shared", "pod_exc", false, []string{"3-4"}},
+	{"pod_exc", "pod_excl_two_container", false, []string{"3-4", "5-7"}},
+	{"pod_exc", "pod_two_container_sh_exc", false, []string{"8", "9-12,15,17"}},
+	{"pod_exc", "pod_three_cont_sh_exc_def", false, []string{"3", "9-12,15,17", "0-2"}},
+	{"pod_exc", "pod_default_explicit", false, []string{"0-2"}},
+	{"pod_exc", "pod_default_implicit", false, []string{"0-2"}},
+	{"pod_exc", "pod_exc_pin_2_proc_2_cont", false, []string{"12-13", "14,16"}},
+	{"pod_exc", "pod_exc_pin_2_proc_1_cont", false, []string{"6"}},
+	{"pod_exc", "pod_pin_2_proc_exc_shared", false, []string{"9-12,15-17"}},
+	{"pod_exc", "pod_not_running", false, []string{"E"}},
+	{"pod_exc", "nodename_missing", false, []string{"E"}},
+	{"pod_exc", "pod_not_ready", false, []string{"E"}},
+	{"pod_exc", "chckpnt_no_device", false, []string{"0-2"}},
+	{"pod_exc", "chckpnt_no_res", false, []string{"0-2"}},
+	{"pod_exc", "chckpnt_no_device_no_res", false, []string{"0-2"}},
+	{"pod_exc", "no_chckpnt_entry", false, []string{"0-2"}},
+	{"pod_exc", "missing_objmeta_uid", true, nil},
+	{"pod_exc", "bad_deviceID_format", false, []string{"E"}},
+	{"pod_exc", "no_cpuset_file", true, nil},
+	{"pod_exc", "no_CID", true, nil},
+	{"pod_exc", "naming_mismatch", false, []string{"E"}},
+}
+
+func TestCreateController(t *testing.T) {
+	testSh := &sethandler.SetHandler{
+		PoolConfig: testPoolConf1,
+		CpusetRoot: getFakeCpusetRoot(wildcardFakeCpuSetPath),
+		K8sClient:  fake.NewSimpleClientset(),
+	}
+	ctrl := testSh.CreateController()
+	if ctrl == nil {
+		t.Errorf("Controller creation failed")
+	}
+}
+
+func TestNew(t *testing.T) {
+	sh, err := sethandler.New(testKubeconfPath, testPoolConf1, getFakeCpusetRoot(wildcardFakeCpuSetPath))
+	if err != nil || sh == nil {
+		t.Errorf("Sethandler object creation failed because: %s", err.Error())
+	}
+}
+
+func TestPodAdded(t *testing.T) {
+	testSethandler, err := setupEnv(testPoolConf1)
+	if err != nil {
+		t.Errorf("Test suite setup failed: %s", err.Error())
+	}
+
+	for _, tc := range podAddedTcs {
+		t.Run(tc.podName, func(t *testing.T) {
+			if tc.podName != "pod_default_explicit_no_default_pool" {
+				testSethandler.PodAdded(getPod(tc.podName))
+				cpusActual, err := readFakeCpusetFile(getPod(tc.podName))
+				if err != nil && !tc.isErrorExpectedAtFakeFileRead {
+					t.Logf("Could not process FAKE cpuset file because: %s", err.Error())
+				}
+				if !reflect.DeepEqual(cpusActual, tc.expectedCpus) {
+					t.Errorf("Mismatch in expected (%s) vs actual (%s) cpus written in cpuset file: ", tc.expectedCpus, cpusActual)
+				}
+			} else {
+				err = utils.RemoveTempSysFs()
+				if err != nil {
+					t.Logf("Removal of temp fs for cpusets was unsuccessful due to: %s", err.Error())
+				}
+				testSethandler2, err := setupEnv(testPoolConf2)
```

Comment:
if you would do environment setup + teardown on the test case level, and based on the new config param you wouldn't need case specific trickery

Reply:
because the rest of the code in this branch is copy-paste basically

Reply:
removed copypaste branch, resolved in the suggested way